contracts/contracts/vault/OETHBaseZapper.sol

```diff
@@ -0,0 +1,134 @@
+     * Will verify that the user is sent 1:1 for ETH.
+     */
+    receive() external payable {
+        deposit();
+    }
+
+    /**
+     * @dev Deposit ETH and receive superOETHb in return
+     * Will verify that the user is sent 1:1 for ETH.
+     * @return Amount of OETH sent to user
+     */
+    function deposit() public payable returns (uint256) {
+        uint256 balance = address(this).balance;
+
+        emit Zap(msg.sender, ETH_MARKER, balance);
+
+        // Wrap ETH
+        weth.deposit{ value: balance }();
+
+        // Mint with WETH
+        return _mint(balance, msg.sender);
+    }
+
+    /**
+     * @dev Deposit ETH and receive superOETHb in return
+     * @param minReceived min amount of wsuperOETHb to receive
+     * @return Amount of wsuperOETHb sent to user
+     */
+    function depositETHForWrappedTokens(uint256 minReceived)
+        external
+        payable
+        returns (uint256)
+    {
+        uint256 balance = address(this).balance;
+
+        emit Zap(msg.sender, ETH_MARKER, balance);
+
+        // Wrap ETH
+        weth.deposit{ value: balance }();
+
+        // Mint with WETH
+        uint256 mintedOethb = _mint(balance, address(this));
+
+        // Wrap superOETHb into wsuperOETHb
+        uint256 mintedWoethb = woethb.deposit(mintedOethb, msg.sender);
+
+        require(mintedWoethb >= minReceived, "Zapper: not enough minted");
+
+        return mintedWoethb;
+    }
+
+    /**
+     * @dev Deposit WETH and receive superOETHb in return
+     * @param wethAmount Amount of WETH to deposit
+     * @param minReceived min amount of wsuperOETHb to receive
+     * @return Amount of wsuperOETHb sent to user
+     */
+    function depositWETHForWrappedTokens(
+        uint256 wethAmount,
+        uint256 minReceived
+    ) external returns (uint256) {
+        // slither-disable-next-line unchecked-transfer unused-return
+        weth.transferFrom(msg.sender, address(this), wethAmount);
+
+        emit Zap(msg.sender, address(weth), wethAmount);
+
+        // Mint with WETH
+        uint256 mintedOethb = _mint(wethAmount, address(this));
+
+        // Wrap superOETHb into wsuperOETHb
+        uint256 mintedWoethb = woethb.deposit(mintedOethb, msg.sender);
+
+        require(mintedWoethb >= minReceived, "Zapper: not enough minted");
+
+        return mintedWoethb;
+    }
+
+    /**
+     * @dev Internal function to mint superOETHb with WETH
+     * @param minOETH Minimum amount of OETH to for user to receive
+     * @param recipient Address that receives the tokens
+     * @return Amount of OETH sent to user
+     */
+    function _mint(uint256 minOETH, address recipient)
+        internal
+        returns (uint256)
+    {
+        uint256 toMint = weth.balanceOf(address(this));
+        vault.mint(address(weth), toMint, minOETH);
+        uint256 mintedAmount = oethb.balanceOf(address(this));
+        require(mintedAmount >= minOETH, "Zapper: not enough minted");
+
+        if (recipient != address(this)) {
```

Comment:
The `recipient` is always `address(this)`. We can remove the `address recipient` parameter from the function and this branching logic since the `oethb` transfer will never happen as a part of this function